File: FrontEnd/src/app/models/task-dto.ts
import { TaskItemStatus } from "./task-item-status";

export interface TaskDto {
    id: number;
    name: string;
    description: string;
    status: TaskItemStatus;
    startDate: Date;
    endDate: Date;
    createdByUserId: string;
    assignedToUserId: string | null;
}

File: BackEnd/Source/TaskManager/TaskManager.Application/DTOs/TaskDto.cs
using TaskManager.Domain.Entities.Enums;

namespace TaskManager.Application.DTOs
{
    public record TaskDto
    (
        int Id,
        string Name,
        string Description,
        TaskItemStatus Status,
        DateTime StartDate,
        DateTime EndDate,
        string CreatedByUserId,
        string? AssignedToUserId
    );
}